### PR TITLE
manifest: add `features` field

### LINKED ISSUE #1936: Add features extension to manifest
See neo-project/neo#2440.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1936 .

### CODE REVIEW

File: pkg/smartcontract/manifest/manifest.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ type Manifest struct {
 	Name string `json:"name"`
 	// ABI is a contract's ABI.
 	ABI ABI `json:"abi"`
+	// Features is a set of contract features. Currently unused.
+	Features map[string]interface{} `json:"features"`
```
Comment (AnnaShaleva) [inline pkg/smartcontract/manifest/manifest.go:36]: Could you, please, store it as `json.RawMessage` to preserve the order of JSON fields? It's the same problem as the one mentioned in https://github.com/neo-project/neo/issues/2181 and https://github.com/neo-project/neo/issues/2345.

File: pkg/smartcontract/manifest/manifest.go [hunk 2]
```diff
@@ -155,6 +158,7 @@ func (m *Manifest) ToStackItem() (stackitem.Item, error) {
 	return stackitem.NewStruct([]stackitem.Item{
 		stackitem.Make(m.Name),
 		stackitem.Make(groups),
+		stackitem.NewMap(),
```
Comment (AnnaShaleva) [inline pkg/smartcontract/manifest/manifest.go:161]: We may need to add a comment to neo-project/neo#2345 about the map representation of `features` field. `stackitem.Map` requires preserving the order of fields, but JSON specification can't provide it. Thus, we need `Manifest.Features` either to be `stackitem.ByteArray` (the same solution was applied in https://github.com/neo-project/neo/pull/2254 to Manifest.Extra) or to have strictly-defined structure.

Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/manifest.go:161]: However, this can wait until neo will support non-empty `features`.

**Review by AnnaShaleva — APPROVED** [review]

**Comment by AnnaShaleva** [comment]
Has something happened with the tests?

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**